src/Pager/PaginationInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Pager;
+
+use Countable;
+use Traversable;
+
+/**
+ * This interface describes the methods for pagination instance.
+ */
+interface PaginationInterface extends Countable, Traversable
+{
+    public function currentPage(): int;
+
+    public function perPage(): int;
+
+    public function totalCount(): ?int;
+
+    public function pageCount(): ?int;
+
+    public function hasPrevPage(): bool;
+
+    public function hasNextPage(): bool;
+
+    public function items(): iterable;
```

Comment:
No, the `PaginatedResult` class used by default extends `IteratorIterator`, so they can directly iterate it without having to use the `items()` method. There are already tests which iterate the result of `Controller::paginate()` which remain unchanged.

I have intentionally just not made `PaginatedResultSet` implement the `CollectionInterface` because I think it's uncommon to do further operations on paginated results. Users always have the convenience of `collection()` if needed.